lib/workers/repository/process/vulnerabilities.ts

```diff
@@ -0,0 +1,142 @@
+    packageFiles: Record<string, PackageFile[]>
+  ): Promise<void> {
+    const managers = Object.keys(packageFiles).filter(
+      (manager) => Vulnerabilities.managerEcosystemMap[manager] !== undefined
+    );
+    const allManagerJobs = managers.map((manager) =>
+      this.fetchManagerVulnerabilities(config, packageFiles, manager)
+    );
+    await Promise.all(allManagerJobs);
+  }
+
+  private async fetchManagerVulnerabilities(
+    config: RenovateConfig,
+    packageFiles: Record<string, PackageFile[]>,
+    manager: string
+  ): Promise<void> {
+    const managerConfig = getManagerConfig(config, manager);
+    const queue = packageFiles[manager].map(
+      (pFile) => (): Promise<void> =>
+        this.fetchManagerPackagerFileUpdates(config, managerConfig, pFile)
+    );
+    logger.trace(
+      { manager, queueLength: queue.length },
+      'fetchManagerUpdates starting'
+    );
+    await pAll(queue, { concurrency: 5 });
+    logger.trace({ manager }, 'fetchManagerUpdates finished');
+  }
+
+  private async fetchManagerPackagerFileUpdates(
+    config: RenovateConfig,
+    managerConfig: RenovateConfig,
+    pFile: PackageFile
+  ): Promise<void> {
+    const { packageFile } = pFile;
+    const packageFileConfig = mergeChildConfig(managerConfig, pFile);
+    const { manager } = packageFileConfig;
+    const queue = pFile.deps.map(
+      (dep) => (): Promise<PackageRule[]> =>
+        this.fetchDependencyVulnerabilities(packageFileConfig, dep)
+    );
+    logger.trace(
+      { manager, packageFile, queueLength: queue.length },
+      'fetchManagerPackagerFileUpdates starting with concurrency'
+    );
+
+    config.packageRules?.push(
+      ...(await pAll(queue, { concurrency: 5 })).flat()
+    );
+    logger.trace({ packageFile }, 'fetchManagerPackagerFileUpdates finished');
+  }
+
+  private async fetchDependencyVulnerabilities(
+    packageFileConfig: RenovateConfig & PackageFile,
+    packageDependency: PackageDependency
+  ): Promise<PackageRule[]> {
+    const ecosystem =
+      Vulnerabilities.managerEcosystemMap[packageFileConfig.manager];
+
+    const vulnerabilities = await this.osvOffline?.getVulnerabilities(
+      ecosystem,
+      packageDependency.depName
+    );
+    return this.convertToPackageRule(
+      vulnerabilities ?? [],
+      packageDependency.depName,
+      ecosystem
+    );
+  }
+
+  private convertToPackageRule(
+    vulnerabilities: Osv.Vulnerability[],
+    dependencyName: string,
+    ecosystem: Ecosystem
+  ): PackageRule[] {
+    return vulnerabilities
+      .flatMap((vulnerability) => vulnerability.affected)
+      .filter(
+        (vulnerability) =>
+          vulnerability?.package?.name === dependencyName &&
+          vulnerability?.package?.ecosystem === ecosystem
+      )
+      .map(
+        (affected): PackageRule => ({
+          matchPackageNames: [dependencyName],
+          allowedVersions: affected?.ranges?.[0].events.find(
+            (event) => event.fixed !== undefined
```

Comment:
```suggestion
            (event) => !!event.fixed
```
shorter code?

Reply:
Less readable in my opinion. I'll leave it as-is, unless you feel strongly about it.